src/web/app/pages-admin/admin-search-page/admin-search-page.component.html

```diff
@@ -149,18 +149,22 @@
                 </li>
                 <li class="list-group-item list-group-item-info">
                   <strong>Course Join Link</strong>
+                  <button type="button" class="btn btn-primary margin-left-5px btn-sm" (click)="openEmail(student.courseId, student.email, EmailType.STUDENT_COURSE_JOIN)">Send mail</button>
                   <input [value]="student.courseJoinLink" readonly class="form-control">
                 </li>
                 <li class="list-group-item list-group-item-warning" *ngFor="let openFs of student.openSessions | keyvalue">
                   <strong>{{ openFs.key }}</strong>
+                  <button type="button" class="btn btn-primary margin-left-5px btn-sm" (click)="openEmail(student.courseId, student.email, EmailType.FEEDBACK_SESSION_REMINDER, sessionLinkToSessionName(openFs.value))">Send mail</button>
```

Comment:
Do we really need `sessionLinkToSessionName` function? There might be something already usable in `openFs`.

Reply:
I think we need it.

`openFs.key` is something like `s2 Sun, 01 Apr yyyy, 11:59 PM +08:00-Sat, 09 Apr yyyy, 11:59 PM +08:00`

`openFs.value` is `http://localhost:4200/web/sessions/submission?courseid=c1&key=8A005B3F3B25C093B205568073DC29A394871A2B4E4D2B30C5725D92FA70F5689F5DB03F086A69415AA0B122AE5F7A05&studentemail=charles%40example.com&fsname=s2`

Either way we'll have to extract it from either of them. I think the URL is safer?

Reply:
I just checked code again. Let's take this time to also improve the implementation of this part; `key` being something like that is not exactly the best way. Better way would be the feedback session (name, open time, close time) to be accessible and the HTML file just needs to call one of those fields as needed.

Also, that `yyyy` is supposed to be the year. That needs to be fixed.